File: Serene/Serene.Script/Imports/ServerImports/Northwind.CustomerForm.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.Northwind
{
    public partial class CustomerForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Customer";

        public CustomerForm(string idPrefix) : base(idPrefix) {}

        public StringEditor CustomerID { [InlineCode("{this}.w('CustomerID', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public StringEditor CompanyName { [InlineCode("{this}.w('CompanyName', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public StringEditor ContactName { [InlineCode("{this}.w('ContactName', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public StringEditor ContactTitle { [InlineCode("{this}.w('ContactTitle', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public LookupEditor Representatives { [InlineCode("{this}.w('Representatives', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public StringEditor Address { [InlineCode("{this}.w('Address', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public StringEditor City { [InlineCode("{this}.w('City', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public StringEditor Region { [InlineCode("{this}.w('Region', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public StringEditor PostalCode { [InlineCode("{this}.w('PostalCode', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public StringEditor Country { [InlineCode("{this}.w('Country', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public StringEditor Phone { [InlineCode("{this}.w('Phone', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public StringEditor Fax { [InlineCode("{this}.w('Fax', Serenity.CodeGeneration.ExternalType)")] get; private set; }
        public NotesEditor NoteList { [InlineCode("{this}.w('NoteList', Serenity.CodeGeneration.ExternalType)")] get; private set; }
    }
}

File: Serene/Serene.Web/Imports/ServerTypings/Northwind.CustomerForm.ts
namespace Serene.Northwind {
    export interface CustomerForm {
        CustomerID: Serenity.StringEditor;
        CompanyName: Serenity.StringEditor;
        ContactName: Serenity.StringEditor;
        ContactTitle: Serenity.StringEditor;
        Representatives: Serenity.LookupEditor;
        Address: Serenity.StringEditor;
        Country: Serenity.LookupEditor;
        City: Serenity.LookupEditor;
        Region: Serenity.StringEditor;
        PostalCode: Serenity.StringEditor;
        Phone: Serenity.StringEditor;
        Fax: Serenity.StringEditor;
        NoteList: NotesEditor;
        LastContactDate: Serenity.DateEditor;
        LastContactedBy: Serenity.LookupEditor;
        Email: Serenity.EmailEditor;
        SendBulletin: Serenity.BooleanEditor;
    }

    export class CustomerForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Customer';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!CustomerForm.init)  {
                CustomerForm.init = true;

                var s = Serenity;
                var w0 = s.StringEditor;
                var w1 = s.LookupEditor;
                var w2 = NotesEditor;
                var w3 = s.DateEditor;
                var w4 = s.EmailEditor;
                var w5 = s.BooleanEditor;

                Q.initFormType(CustomerForm, [
                    'CustomerID', w0,
                    'CompanyName', w0,
                    'ContactName', w0,
                    'ContactTitle', w0,
                    'Representatives', w1,
                    'Address', w0,
                    'Country', w1,
                    'City', w1,
                    'Region', w0,
                    'PostalCode', w0,
                    'Phone', w0,
                    'Fax', w0,
                    'NoteList', w2,
                    'LastContactDate', w3,
                    'LastContactedBy', w1,
                    'Email', w4,
                    'SendBulletin', w5
                ]);
            }
        }
    }
}

